src/management/projects.ts

```diff
@@ -94,41 +152,60 @@ export async function addFirebaseToCloudProject(projectId: string): Promise<any>
 }
 
 /**
- * Lists all Firebase projects associated with the currently logged-in account. Repeatedly calls the
- * paginated API until all pages have been read.
- * @return a promise that resolves to the list of all projects.
+ * Lists Firebase projects in a page using the paginated API, identified by the page token and its
+ * size.
  */
-export async function listFirebaseProjects(
-  pageSize: number = PROJECT_LIST_PAGE_SIZE
-): Promise<FirebaseProjectMetadata[]> {
-  const projects: FirebaseProjectMetadata[] = [];
+export async function getProjectPage(
+  pageSize: number = PROJECT_LIST_PAGE_SIZE,
+  pageToken?: string
+): Promise<FirebaseProjectPage> {
+  let apiResponse;
+
   try {
-    let nextPageToken = "";
-    do {
-      const pageTokenQueryString = nextPageToken ? `&pageToken=${nextPageToken}` : "";
-      const response = await api.request(
-        "GET",
-        `/v1beta1/projects?pageSize=${pageSize}${pageTokenQueryString}`,
-        {
-          auth: true,
-          origin: api.firebaseApiOrigin,
-          timeout: TIMEOUT_MILLIS,
-        }
-      );
-      if (response.body.results) {
-        projects.push(...response.body.results);
+    const pageTokenQueryString = pageToken ? `&pageToken=${pageToken}` : "";
+    apiResponse = await api.request(
+      "GET",
+      `/v1beta1/projects?pageSize=${pageSize}${pageTokenQueryString}`,
+      {
+        auth: true,
+        origin: api.firebaseApiOrigin,
+        timeout: TIMEOUT_MILLIS,
       }
-      nextPageToken = response.body.nextPageToken;
-    } while (nextPageToken);
-
-    return projects;
+    );
   } catch (err) {
     logger.debug(err.message);
     throw new FirebaseError(
       "Failed to list Firebase projects. See firebase-debug.log for more info.",
       { exit: 2, original: err }
     );
   }
+
+  const projectPage: FirebaseProjectPage = { projects: [] };
+  if (apiResponse.body.results) {
+    projectPage.projects.push(...apiResponse.body.results);
```

Comment:
I don't think we actually need to do an array `push` here, since these are only the results from a single page. Also we might not need the `if` statements. Does the below code work?

```ts
const projectPage: FirebaseProjectPage = {
  projects: apiResponse.body.results || [],
  nextPageToken: apiResponse.body.nextPageToken,
};
return projectPage;
```

Reply:
Yes, I think this is better here. Updated